src/WebSocketHandler.js

```diff
@@ -5,246 +5,35 @@
-
-    // Date of latest setup call. The setup is the way to open a new connection.
-    this.latestSetupDate = null;
-
-    // Date of latest ping.
-    this.latestPingDate = null;
-
-    // Latest round trip time measured by PING/PONG.
-    this.latestRTT = null;
-
-    // Timer used to detected when connection is down.
-    this.timeoutTimer = null;
-  }
-
-  /**
-   * Start websocket object and its methods
-   */
-  setup() {
-    if (this.started) {
-      return;
-    }
-    let wsURL = helpers.getWSServerURL();
-    if (wsURL === null) {
-      return;
-    }
-
-    if (this.ws) {
-      const dt = (new Date() - this.latestSetupDate) / 1000;
-      if (dt < this.openConnectionTimeout) {
-        return;
-      }
-      this.ws.close();
-    }
-    this.ws = new this.WebSocket(wsURL);
-    this.latestSetupDate = new Date();
-
-    this.ws.onopen = () => {
-      this.onOpen();
-    }
-    this.ws.onmessage = (evt) => {
-      this.onMessage(evt);
-    }
-    this.ws.onerror = (evt) => {
-      this.onError(evt);
-    }
-    this.ws.onclose = () => {
-      this.onClose();
-    }
-  }
-
-  onPong() {
-    if (this.latestPingDate) {
-      const dt = (new Date() - this.latestPingDate) / 1000;
-      this.latestRTT = dt;
-      this.latestPingDate = null;
-    }
-    if (this.timeoutTimer) {
-      clearTimeout(this.timeoutTimer);
-      this.timeoutTimer = null;
-    }
-  }
-
-  /**
-   * Handle message receiving from websocket
-   *
-   * @param {Object} evt Event that has data (evt.data) sent in the websocket
-   */
-  onMessage(evt) {
-    const message = JSON.parse(evt.data)
-    const _type = message.type.split(':')[0]
-    if (_type === 'pong') {
-      this.onPong();
-    }
-    this.emit(_type, message)
-  }
-
-  /**
-   * Method called when websocket connection is opened
-   */
-  onOpen() {
-    if (this.connected === false) {
-      // If was not connected  we need to reload data
-      // Emits event to reload data
-      this.emit('reload_data');
-    }
-    this.connected = true;
-    this.started = true;
-    this.setIsOnline(true);
-    this.heartbeat = setInterval(() => {
-      this.sendPing();
-    }, this.heartbeatInterval);
-    wallet.onWebsocketOpened();
-  }
-
-  /**
-   * Method called when websocket connection is closed
-   */
-  onClose() {
-    this.started = false;
-    this.connected = false;
-    this.setIsOnline(false);
-    wallet.onWebsocketBeforeClose();
-    if (this.ws) {
-      this.ws.close();
-      this.ws = null;
-    }
+import helpers from './helpers';
+import WS from './websocket';
+
+class WebSocketHandler {
+  constructor() {
+    this.ws = null;
+    /* Right after importing the modules helpers and wallet, they are
+     * not available and are still undefined.
+     * This is probably caused by a cyclic import (wallet -> helpers -> tokens -> wallet)
+     * but still need more study.
+     * For now the setTimeout is used, so we can use the helpers module properly.
+     */
```

Comment:
With the current solution for `WebSocketHandler` and `Storage` we won't need to refactor any wallet that uses the lib. I think we should create an issue to handle this after, so it's a small refactor, rather than just mix everything here. What do you think?

Reply:
We don't need to update the wallets right after this change. 

Let's say we move these functions as I suggested. Next time we have to update a wallet and use this new version of the lib, we'll get an error as there won't be `getDepositAmount` and `getWithdrawAmount` in `helpers.js`. We'll only need to change where we import them from. Seems like a simple really change to me and we don't need this hack here.

Reply:
I've removed the cyclic import and also the `setTimeout` hack. When I test this version in the wallets I just need to change where we get `getDepositAmount` and `getWithdrawAmount`